### DISCUSSION TITLE
Conventions and whitepaper

### DISCUSSION THREAD

**Discussion by Hjaf** [microsoft/Power-Fx#321] [Q&A] [answered]
I see that some of the examples in the documentation use slightly different conventions for naming of variables etc. Not sure if the [coding standards and guidelines(whitepaper)](https://aka.ms/powerappscanvasguidelines) have been kept up to date since the [blog post](https://powerapps.microsoft.com/en-us/blog/powerapps-canvas-app-coding-standards-and-guidelines/) that introduced it was published in 2018, or if an update is being developed. 

I find the whitepaper conventions and guidelines to be useful for more complex apps, is someone keeping this up to date with the Power FX roadmap? What are your recommendations concerning code conventions in Power FX?

**Comment by MikeStall** [comment, marked as answer]
The whitepaper there looks like it's targeted at Power Apps (Canvas apps).  That has to deal with things like controls, screens, etc - concepts that are in a canvas app and not just the expression engine. Power Fx is "just" the expression engine but it is used very broadly across multiple products. We don't have a similar whitepaper for just power fx conventions. In other words, there's currently no "Style Cop" for power fx - although part of the goal is to keep power fx simple enough that it doesn't require a style cop to be maintainable.

**Reply by Hjaf** [reply]
Thanks for your answer. I suspected as much, I just noticed many of the examples appear to be from PowerApps Studio and wanted to hear your thoughts on this subject :) 

The whitepaper conventions for power apps development try to make it easier to see scope and control/type in the formula alone, this quickly becomes a challenge in Power Apps (canvas).
